### PR TITLE
Feat/api/fee account

### PR DESCRIPTION
**Pull request by frontr-uk**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/auth/index.ts [hunk 1]
```diff
@@ -94,10 +96,10 @@ export async function oauth(req: EnhancedRequest, res: express.Response, next: e
             res.cookie(config.cookies.token, accessToken)
 
             const orgIdResponse = await getOrganisationId(details)
-            const orgId = orgIdResponse.data.id
-
+            // const orgId = orgIdResponse.data.id
+            //
             session.auth = {
-                orgId,
+                orgId: orgIdResponse.data.id,
```
Comment (fazanki) [inline api/auth/index.ts]: why did you change this?  use one or another

File: src/fee-accounts/services/fee-accounts.service.ts [hunk 2]
```diff
@@ -6,35 +6,39 @@ import {map} from '../../../node_modules/rxjs/operators';
...
 @Injectable()
 export class FeeAccountsService {
   constructor(private http: HttpClient) {
   }
 
-  fetchFeeAccounts(): Observable<any> {
-    return of(feeAccountsDummy);
+  fetchFeeAccounts(): Observable<Array<PbaAccounts>> {
+    // return of(feeAccountsDummy);
+    // const tempId = 'b4775ea1-4036-4d7b-bebd-0b7cdc3c786f'
```
Comment (fazanki) [inline src/fee-accounts/services/fee-accounts.service.ts]: do we need this?

File: src/fee-accounts/services/fee-accounts.service.ts [hunk 3]
```diff
@@ -6,35 +6,39 @@ import {map} from '../../../node_modules/rxjs/operators';
...
+    //     case: item.case_reference,
+    //     reference: 'NO DATA to MAP',
+    //     submittedBy: 'NO DATA to MAP',
+    //     status: 'NO DATA to MAP',
+    //     dateCreated: item.date_created,
+    //     amount: item.amount,
+    //     dateUpdated: item.date_updated,
+    //     routerLink: `account/${item.account_number}/summary`
+    //   };
+    // }));
```
Comment (fazanki) [inline src/fee-accounts/services/fee-accounts.service.ts]: do we need this

File: src/fee-accounts/store/reducers/single-fee-account.reducer.ts [hunk 4]
```diff
@@ -40,7 +40,7 @@ export function reducer(
       return {
         ...state,
         ...{
-          overview:{
+          overview: {
```
Comment (fazanki) [inline src/fee-accounts/store/reducers/single-fee-account.reducer.ts]: same here remove ... and {

File: src/fee-accounts/store/reducers/single-fee-account.reducer.ts [hunk 5]
```diff
@@ -54,11 +54,13 @@ export function reducer(
       console.log('fromSingleFeeAccountActions payload', JSON.stringify(payload));
       return {
         ...state,
-        ...{transactions:{
-          data: action.payload,
-          loaded: true,
-          loading: false
-        }}
+        ...{
```
Comment (fazanki) [inline src/fee-accounts/store/reducers/single-fee-account.reducer.ts]: same here remove ...{

File: src/fee-accounts/store/selectors/fee-accounts.selectors.ts [hunk 6]
```diff
@@ -1,16 +1,36 @@
...
+export const getFeeAccountsState = createSelector( selectFeatureFee, (state: any) => state.feeAccounts);
+export const getFeeAccountsRaw = createSelector( getFeeAccountsState, fromFeeAccounts.getFeeAccounts);
+export const getFeeAccountsLoading = createSelector( getFeeAccountsState, fromFeeAccounts.getFeeAccountsLoading);
+export const getFeeAccountsMapped = createSelector(
+  getFeeAccountsRaw,
+  (entities: any) => {
+    const obj = entities.map((entity: any) => {
+      const element = {
+          ...entity,
+          routerLink: `/fee-accounts/account/${entity.pbaNumber}/summary`
```
Comment (fazanki) [inline src/fee-accounts/store/selectors/fee-accounts.selectors.ts]: i think this should happen in the reducer and get storred in the state

File: src/fee-accounts/store/reducers/fee-accounts.reducer.ts [hunk 7]
```diff
@@ -7,7 +7,7 @@ export interface FeeAccountsState {
 }
 
 export const initialState: FeeAccountsState = {
-  feeAccounts: [],
+  feeAccounts: [{}],
```
Comment (fazanki) [inline src/fee-accounts/store/reducers/fee-accounts.reducer.ts]: d u need an empty object in the array ?

File: src/fee-accounts/containers/overview/account-overview.component.ts [hunk 8]
```diff
@@ -12,33 +13,22 @@ export class AccountsOverviewComponent implements OnInit, OnDestroy {
...
-      { header: 'Account name', key: 'accountName' }
+      { header: 'Account number', key: 'pbaNumber', type: 'link' },
+      { header: 'Oraganisation Id', key: 'organisationId' }
     ];
   }
 
-  ngOnDestroy() {
-    this.feeAccountsSubscription.unsubscribe();
-  }
+  ngOnDestroy() {}
```
Comment (fazanki) [inline src/fee-accounts/containers/overview/account-overview.component.ts]: remove if not used

File: src/fee-accounts/containers/overview/account-overview.component.ts [hunk 9]
```diff
@@ -12,33 +13,22 @@ export class AccountsOverviewComponent implements OnInit, OnDestroy {
   columnConfig: GovukTableColumnConfig[];
   tableRows: {}[];
   feeAccountsSubscription: Subscription;
+  accounts$: Observable<Array<any>>;
```
Comment (fazanki) [inline src/fee-accounts/containers/overview/account-overview.component.ts]: maybe would be good to create model for this

File: src/fee-accounts/containers/account-summary/account-summary.component.html [hunk 10]
```diff
@@ -10,5 +10,9 @@ <h1 class="govuk-body govuk-!-font-size-48 govuk-!-font-weight-bold">
   </div>
 </div>
 <app-prd-summary-component
+  *ngIf="(accountSummary$ | async).length === 0 else loading"
```
Comment (fazanki) [inline src/fee-accounts/containers/account-summary/account-summary.component.html]: i think this loading should be hooked on the reducer loading property?

File: src/fee-accounts/containers/account-transactions/account-transactions.component.html [hunk 11]
```diff
@@ -7,5 +7,10 @@ <h1 class="govuk-body govuk-!-font-size-48 govuk-!-font-weight-bold">
   </div>
   <div class="hmcts-page-heading__actions-wrapper"></div>
 </div>
-<!--<pre>{{accountTransactions$ }}</pre>-->
-<app-govuk-table [columnConfig]="columnConfig" [rows]="accountTransactions$ | async" [firstCellIsHeader]="false" [caption]="''"></app-govuk-table>
+<app-govuk-table
+  *ngIf="!(loading$ | async) else loading"
```
Comment (fazanki) [inline src/fee-accounts/containers/account-transactions/account-transactions.component.html]: this should be hooked on the reducer loading property

File: api/auth/index.ts [hunk 12]
```diff
@@ -21,11 +21,12 @@ export async function attach(req: EnhancedRequest, res: express.Response, next:
...
 
         req.headers.ServiceAuthorization = token.token
-
+        logger.info(' session.auth.userId',  session.auth.userId)
         const userId = session.auth.userId
         const jwt = session.auth.token
         const roles = session.auth.roles
-        logger.info('user roles', roles)
+        const orgId = session.auth.orgId
+
```
Comment (fazanki) [inline api/auth/index.ts]: use this org id do not repeat it below

File: api/accounts/data.mock.ts [hunk 13]
```diff
@@ -1,64 +1,65 @@
-export const PaymentMock ={
+import {Payments} from './pba-transactions'
+export const PaymentMock: Payments = {
```
Comment (fazanki) [inline api/accounts/data.mock.ts]: make to do when to remove it

File: api/auth/index.ts (hunk 1 above)
Comment (frontr-uk) [reply api/auth/index.ts]: To keep it in object the Key: Reference , as you had it outside as const.

File: src/fee-accounts/containers/account-summary/account-summary.component.html (hunk 10 above)
Comment (frontr-uk) [reply src/fee-accounts/containers/account-summary/account-summary.component.html]: hmm, I thought I had this updated. Will update

File: src/fee-accounts/containers/account-transactions/account-transactions.component.html (hunk 11 above)
Comment (frontr-uk) [reply src/fee-accounts/containers/account-transactions/account-transactions.component.html]: ? This is hooked on the reducer. it is a selector that get Boolean if the data is loaded or not.

File: src/fee-accounts/containers/overview/account-overview.component.ts (hunk 9 above)
Comment (frontr-uk) [reply src/fee-accounts/containers/overview/account-overview.component.ts]: Good spot.

File: src/fee-accounts/services/fee-accounts.service.ts (hunk 2 above)
Comment (frontr-uk) [reply src/fee-accounts/services/fee-accounts.service.ts]: should be removed from git

File: src/fee-accounts/containers/account-transactions/account-transactions.component.html (hunk 11 above)
Comment (fazanki) [reply src/fee-accounts/containers/account-transactions/account-transactions.component.html]: sorry my bad you are right

File: src/fee-accounts/services/fee-accounts.service.ts (hunk 3 above)
Comment (frontr-uk) [reply src/fee-accounts/services/fee-accounts.service.ts]: doone
